packml_plugin/src/packml_widget.cpp

```diff
@@ -0,0 +1,360 @@
+/**
+ * @license Software License Agreement (Apache License)
+ *
+ * @copyright Copyright (c) 2017 Austin Deric
+ * @copyright Copyright (c) 2019 ROS-Industrial Consortium Asia Pacific (ROS 2 compatibility)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include <iomanip>
+#include <sstream>
+#include "packml_plugin/packml_widget.h"
+#include "rclcpp/rclcpp.hpp"
+#include "packml_msgs/srv/transition.hpp"  // Datatypes for services
+#include "packml_msgs/srv/all_status.hpp"
+
+PackmlWidget::PackmlWidget(QWidget* parent) : QWidget(parent)
+{
+  nh_ = rclcpp::Node::make_shared("uiPlugin");  // Creates a node
+  // UI setup
+  ui_ = new Ui::PackmlPanel;
+  ui_->setupUi(this);
+  QTimer *timer = new QTimer(this);
+  connect(timer, SIGNAL(timeout()), this, SLOT(statusRequest()));
+  timer->start(200);
```

Comment:
Who deletes timer?

Reply:
I use `timerEvent` for this here in https://github.com/ros-industrial/packml_ros2/pull/2/commits/de9d8f837af241c3d71fbf76ababad5f3c3672ec.